src/Elliptic/DiscontinuousGalerkin/SubdomainOperator/InitializeSubdomain.hpp

```diff
@@ -0,0 +1,521 @@
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename... Tags>
+struct TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+/// Actions related to the DG subdomain operator
+namespace elliptic::dg::subdomain_operator::Actions {
+
+namespace detail {
+// Initialize the geometry of a neighbor into which an overlap extends
+template <size_t Dim>
+void initialize_overlap_geometry(
+    const gsl::not_null<Mesh<Dim>*> mesh,
+    const gsl::not_null<size_t*> extruding_extent,
+    const gsl::not_null<Element<Dim>*> element,
+    const gsl::not_null<ElementMap<Dim, Frame::Inertial>*> element_map,
+    const gsl::not_null<tnsr::I<DataVector, Dim>*> inertial_coords,
+    const gsl::not_null<
+        InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
+        inv_jacobian,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::I<DataVector, Dim>>*>
+        face_inertial_coords_internal,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::I<DataVector, Dim>>*>
+        face_inertial_coords_external,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::i<DataVector, Dim>>*>
+        face_normals_internal,
+    const gsl::not_null<
+        std::unordered_map<Direction<Dim>, tnsr::i<DataVector, Dim>>*>
+        face_normals_external,
+    const gsl::not_null<std::unordered_map<Direction<Dim>, Scalar<DataVector>>*>
+    /*face_normal_magnitudes_internal*/,
+    const gsl::not_null<std::unordered_map<Direction<Dim>, Scalar<DataVector>>*>
+    /*face_normal_magnitudes_external*/,
+    const gsl::not_null<::dg::MortarMap<Dim, Mesh<Dim - 1>>*> mortar_meshes,
+    const gsl::not_null<::dg::MortarMap<Dim, ::dg::MortarSize<Dim - 1>>*>
+        mortar_sizes,
+    const gsl::not_null<::dg::MortarMap<Dim, Mesh<Dim>>*> neighbor_meshes,
+    const gsl::not_null<::dg::MortarMap<Dim, Scalar<DataVector>>*>
+    /*neighbor_face_normal_magnitudes*/,
+    const gsl::not_null<::dg::MortarMap<Dim, Mesh<Dim - 1>>*>
+        neighbor_mortar_meshes,
+    const gsl::not_null<::dg::MortarMap<Dim, ::dg::MortarSize<Dim - 1>>*>
+        neighbor_mortar_sizes,
+    const std::vector<std::array<size_t, Dim>>& initial_extents,
+    const std::vector<std::array<size_t, Dim>>& initial_refinement,
+    const Domain<Dim>& domain, const size_t max_overlap,
+    const ElementId<Dim>& element_id,
+    const Direction<Dim>& overlap_direction) noexcept {
+  // Mesh
```

Comment:
while I don't disagree with all these short comments, I'm not sure what value they are adding. Could they be elaborated on someone?

Reply:
Not sure, they're just there to improve readability. Want me to remove them?

Reply:
Whatever you prefer :)